lib/docx/containers/text_run.rb

```diff
@@ -57,6 +57,19 @@ def substitute(match, replacement)
           reset_text
         end
 
+        # Weird things with how $1/$2 in regex blocks are handled means we can't just delegate
+        # block to gsub to get block, we have to do it this way, with a block that gets a MatchData,
+        # from which captures and other match data can be retrieved.
+        # https://ruby-doc.org/3.3.7/MatchData.html
+        def substitute_block(match, &block)

```

Comment:
I've never seen `xxx_block` method in Ruby world. How about to name it `substitute` ?

Reply:
There is already a substitute method with a different signature `def substitute(match, replacement)`

So this one needs a different name.  `substitute_with_block`?  

They could instead be combined into one method? That ignores the `replacement` argument if block is given?

I did not know if that would be considered backward compatible. I worried it would be confusing since the block argument does not work quite like ordinary `gsub` -- becuase ruby does not make that feasible. 

I am happy to name method whatever non-conflicting name maintainers want. Or refactor however wanted for htis functionality. My branch also allows edits by maintainers if you want to edit it yourself I do not mind.